bundles/org.openhab.binding.revogismartstripcontrol/src/main/java/org/openhab/binding/revogismartstripcontrol/internal/udp/DatagramSocketWrapper.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogismartstripcontrol.internal.udp;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DatagramSocketWrapper} wraps Java's DatagramSocket for better testing
+ * UdpSenderService
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class DatagramSocketWrapper {
+
+    private final Logger logger = LoggerFactory.getLogger(DatagramSocketWrapper.class);
+    @Nullable
+    DatagramSocket datagramSocket;
+
+    public void initSocket() throws SocketException {
+        closeSocket();
+        datagramSocket = new DatagramSocket();
+        datagramSocket.setBroadcast(true);
+        datagramSocket.setSoTimeout(3);
+    }
+
+    public void closeSocket() {
+        if (datagramSocket != null && !datagramSocket.isClosed()) {
```

Comment:
In order to prevent the null warnings you need to create a local variable to capture the field instance. Please do the same where ever applicable.
```suggestion
        DatagramSocket datagramSocket = this.datagramSocket;
        if (datagramSocket != null && !datagramSocket.isClosed()) {
```

Reply:
But I want to operate on the objects instance of the DatagramSocket. I will not change that. There is no real NPE problem, because it is checked before usage.